chainer/training/extensions/variable_statistics_plot.py

```diff
@@ -0,0 +1,253 @@
+    percentiles.
+
+    This extension uses reservoir sampling to preserve memory, using a
+    fixed size sample. This means that collected items in the sample are
+    discarded uniformly at random when the number of items becomes larger
+    than the maximum sample size, but each item is expected to occur in the
+    sample with equal probability.
+
+    Args:
+        targets (:class:`Variable`, :class:`Link` or list of either):
+            Parameters for which statistics are collected.
+        max_sample_size (int): Maximum number of samples.
+        report_data (bool):  If ``True``, data statistics are plotted.
+            If ``False``, they are neither computed or plotted.
+        report_grad (bool):  If ``True``, gradient statistics are plotted.
+            If ``False``, they are neither computed or plotted.
+        trigger: Trigger that decides when to save the plots as an image.
+            This is distinct from the trigger of this extension
+            itself. If it is a tuple in the form ``<int>, 'epoch'`` or
+            ``<int>, 'iteration'``, it is passed to :class:`IntervalTrigger`.
+        file_name (str): Name of the plot file under the output directory.
+        figsize (tuple of int): Matlotlib ``figsize`` argument that specifies
+            the size of the output image.
+        marker (str): Matplotlib ``marker`` argument that specified the marker
+            style of the plots.
+        grid (bool): Matplotlib ``grid`` argument that specifies whether grids
+            are rendered in in the plots or not.
+    """
+
+    def __init__(self, targets, max_sample_size=1000,
+                 report_data=True, report_grad=True,
+                 trigger=(1, 'epoch'), file_name='statistics.png',
+                 figsize=None, marker=None, grid=True):
+
+        if file_name is None:
+            raise ValueError('Missing output file name of statstics plot')
+
+        self._vars = _unpack_variables(targets)
+        if len(self._vars) == 0:
+            raise ValueError(
+                'Need at least one variables for which to collect statistics.'
+                '\nActual: 0 <= 0')
+
+        self._report_data = report_data
+        self._report_grad = report_grad
+        self._trigger = trigger_module.get_trigger(trigger)
+
+        self._file_name = file_name
+        self._figsize = figsize
+        self._marker = marker
+        self._grid = grid
+
+        self._keys = []
+        if report_data:
+            self._keys.append('data')
+        if report_grad:
+            self._keys.append('grad')
+
+        self._statistician = Statistician()
+        self._data_shape = (len(self._keys), self._statistician.data_size)
+        self._samples = Reservoir(max_sample_size, data_shape=self._data_shape)
+
+    @staticmethod
+    def available():
+        _check_available()
+        return _available
+
+    def __call__(self, trainer):
+        if not _available:
```

Comment:
If matplotlib is not available, it is better to raise an Exception on `__init__`, isn't it?

Reply:
What do you think about following `plot_report`, raise a warning on `__init__` and then do nothing in `__call__` depending on availability.